cloudnetpy/plotting/plotting.py

```diff
@@ -558,69 +573,6 @@ def plot_2d(data: ndarray, cbar: bool = True, cmap: str = 'viridis',
     plt.show()
 
 
-def _plot_relative_error(ax, error: ndarray, ax_values: tuple):
-    pl = ax.pcolorfast(*ax_values, error[:-1, :-1].T, cmap='RdBu', vmin=-30,
-                       vmax=30)
-    colorbar = _init_colorbar(pl, ax)
-    colorbar.set_label("%", fontsize=13)
-    median_error = ma.median(error.compressed())
-    median_error = "%.3f" % median_error
-    ax.set_title(f"Median relative error: {median_error} %", fontsize=14)
-
-
-def _lin2log(*args: ndarray) -> list:
-    return [ma.log10(x) for x in args]
-
-
-# LEGACY DATA PLOTTING ROUTINES:
-
-
-def generate_legacy_figure(full_path: str,
-                           product: str,
-                           field_name: str,
-                           show: bool = False,
-                           save_path: str = None,
-                           max_y: int = 12,
-                           dpi: int = 200,
-                           image_name: str = None) -> Dimensions:
-    """ Plots one particular field from Cloudnet legacy product file.
-
-    Args:
-        full_path (str): Full path of a legacy Cloudnet file.
-        product (str): Cloudnet file type.
-        field_name (str): Name of variable to be plotted.
-        show (bool, optional): If True, shows the plot. Default is False.
-        save_path (str, optional): If defined, saves the image to this path.
-            Default is None.
-        max_y (int, optional): Upper limit of images (km). Default is 12.
-        dpi (int, optional): Quality of plots. Default is 200.
-        image_name (str, optional): Name (and full path) of the output image.
-            Overrides the *save_path* option. Default is None.
-
-    Returns:
-        Dimensions of the generated figure in pixels.
-
-    """
-    plot_type = ATTRIBUTES[field_name].plot_type
-    field = _find_valid_fields(full_path, [field_name])[0][0]
-    fig, ax = _initialize_figure(1, dpi)
-    ax = ax[0]
-    ax_values = _read_ax_values(full_path, product)
-    field, ax_value = _screen_high_altitudes(field, ax_values, max_y)
-    _set_ax(ax, max_y)
-    if plot_type == 'bar':
-        _plot_bar_data(ax, field, ax_value[0])
-        _set_ax(ax, 2, ATTRIBUTES[field_name].ylabel)
-    elif plot_type == 'segment':
-        field, field_name = legacy_meta.fix_legacy_data(field, field_name)
```

Comment:
Should this still be called in `generate_figure`? If not, maybe `legacy_meta.py` can also be removed?

Reply:
Good point, I think we could remove all legacy-references from the code base. "Legacy" files and Cloudnetpy-generated files are now similar enough, so that we don't need such distinction. I will check this and think about adding a few tests too.